File: src/main/java/org/example/gogoma/controller/response/MarathonSearchResponse.java
package org.example.gogoma.controller.response;

import lombok.Builder;
import lombok.Getter;
import org.example.gogoma.domain.marathon.dto.MarathonPreviewDto;

import java.util.List;

@Getter
@Builder
public class MarathonSearchResponse {

    private List<MarathonPreviewDto> marathonPreviewDtoList;
    private List<String> cityList;
    private List<Integer> marathonTypeList;

    public static MarathonSearchResponse of(
            List<MarathonPreviewDto> marathonPreviewDtoList, List<String> cityList, List<Integer> marathonTypeList) {

        return MarathonSearchResponse.builder()
                .marathonPreviewDtoList(marathonPreviewDtoList)
                .cityList(cityList)
                .marathonTypeList(marathonTypeList)
                .build();
    }
}

File: app/src/main/java/com/example/gogoma/data/model/MarathonSearchResponse.kt
package com.example.gogoma.data.model

import android.os.Parcelable
import kotlinx.parcelize.Parcelize

@Parcelize
data class MarathonSearchResponse(
    val marathonPreviewDtoList: List<MarathonPreviewDto>,
    val cityList: List<String>
) : Parcelable
